### PR TITLE
tests: drivers: i3c: Add comprehensive I3C driver test suite

### PR DESCRIPTION
**Pull request by sudiiyengar-en**
Adds a multi-test-suite I3C driver validation framework supporting discovery, CCC commands, IBI, negative tests, stress tests, and sensor integration.

Test suites:
- Discovery: DAA, RSTDAA, PID retrieval, target enumeration
- CCC: GETPID, GETBCR, GETDCR, SETMWL, GETMWL, GETMRL, GETSTATUS, ENEC/DISEC event control, broadcast commands
- IBI: In-band interrupt capability detection and validation
- Negative: Error handling, invalid addresses, boundary conditions
- Stress: Repeated operations (10x GETPID), multi-target validation
- Sensor: Real device tests for BMI323 and ICM42670 sensors

Features:
- Multi-speed mode testing (SDR, I2C-FM+)
- RSTDAA+DAA reset sequence per test for clean state
- Device tree based target configuration via overlay
- Macro-based test iteration across modes and targets

Hardware requirements:
- I3C controller (i3c0)
- BMI323 sensor (default) or ICM42670 (optional)

### CODE REVIEW

File: tests/drivers/i3c/Kconfig [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	  This option enables the sensor test suite which validates
+	  real I3C sensor devices including the BMI323 accelerometer
+	  and gyroscope, and the ICM42670 motion sensor. These tests
+	  require the actual sensor hardware to be connected and
+	  configured in the device tree. Enable this only when testing
+	  with physical sensor devices present on the I3C bus.
+
+endmenu
+
+source "Kconfig.zephyr"
```
[the remark this answers is not in the document]
Comment (sudiiyengar-en) [reply tests/drivers/i3c/Kconfig:85]: This is needed other compilation error will come

File: tests/drivers/i3c/README.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+.. code-block:: console
+
+   west twister -T tests/drivers/i3c --list-tests
+
+Run tests:
+
+.. code-block:: console
+
+   west twister -T tests/drivers/i3c -p alif_e8_dk/ae822fa0e5597xx0/rtss_hp
+
```
Comment (VeijoPesonen) [inline tests/drivers/i3c/README.rst]: Thanks for the instructions but how to run tests is covered by Zephyr's own documentation. No need to do the same work twice.
```suggestion
```

Comment (sudiiyengar-en) [reply tests/drivers/i3c/README.rst]: Thanks updated

**Review by AnttiKauppila-AlifSemi — APPROVED** [review]

**Review by VeijoPesonen — APPROVED** [review]
